src/components/docs/WaveBackground/index.js

```diff
@@ -1,18 +1,44 @@
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 import styles from './styles.module.css';
 
 import useThemeContext from '@theme/hooks/useThemeContext';
 
+const ID = "wave-background";
+
+const setWaveHeight = setHeight => {
+  const container = document.querySelector('main');
+  const image = document.querySelector(`#${ID} img`);
+  const wave = document.getElementById(ID);
+
+  const totalHeight = container.clientHeight;
+  const imageHeight = image.clientHeight;
+  const distanceFromTop = wave.offsetTop;
+  const internalOffset = eval(styles['total-offset']);
+
+  setHeight(totalHeight - distanceFromTop - imageHeight - internalOffset);
+};
+
 const WaveBackground = () => {
   const {isDarkTheme} = useThemeContext();
+  const [height, setHeight] = useState(0);
 
   const url = isDarkTheme ? '/img/docs/wave-top-dark.svg' : '/img/docs/wave-top.svg';
 
+  useEffect(() => {
+    setWaveHeight(setHeight);
```

Comment:
Are you just tryi g to trigger functionality on window resize?

Reply:
Yes, but more than that. Window resize is one thing that will change the total height of the body, but there are other things too. For example, if you hover over a different tab on the fancy code snippet going from a short snippet to a longer one, that would change the height of the code snippet component, which in turn would change the height of the overall body. 
Therefore, the resize window observer is not sufficient. The `ResizeObserver` was created for exactly this purpose